### DISCUSSION TITLE
PARTIAL sync (ID-based) instead of FULL sync

### DISCUSSION THREAD

**Discussion by JiriOgurek** [Mimetis/Dotmim.Sync#646] [Q&A]
First of all, big THANK YOU for complex, straightforward and easy to setup sync framework, great job! I am using it for one-way (UploadOnly), insert only (removing delete changes), partial/selection (based on ID) sync from client to server. It works perfect for the first sync for selected ID, however at the second attempt for the next ID (existing already at the time of first sync), nothing gets synced. I guess used filter deletes all changes except filtered ID at the first run, how can I preserve these changes for the second run. My goal is to perform partial sync (for selected ID only) instead of full sync, when all outfiltered changes are deleted. Thank you for your kind help :-)

**Comment by Mimetis** [comment]
Never change a SyncParameter value for one client database
If you want to change your filter value, use a SyncType.Reinitialize to reinitialize the client database with your new filter

**Reply by JiriOgurek** [reply]
Thank you for your reply. So if I want to perform partial sync only based on ID, I have to set Reinitialize for each particular ID filter, right?

**Reply by Mimetis** [reply]
Can you explain a little bit more your scenario ?
Why do you need to change a SyncParameter value for a particular client database ?

**Comment by JiriOgurek** [comment]
Sure, I am using your framework to upload particular dataset (based in ID) from the client to the server only. For the first upload went all good, but for the second upload of the different ID set there was nothing to upload. How can I do it right, please?

**Reply by Mimetis** [reply]
You can use scopes I guess
Make a scope (named A) with tables and 1 filter parameter
Then make a scope (named B) with tables and 1 other filter parameter

Never tested, but it worth a try

**Reply by JiriOgurek** [reply]
Thank you for your kind hint. I've tried to do so, looking at Multi scopes sample, however there's non-existing (outdated?) constructor used: 
var setupProducts = new SyncSetup(productScopeTables, "productScope");

How can I make my own scopes as you suggested, please?

**Reply by JiriOgurek** [reply]
I've noticed that scope has been moved from setup to agent constructor. Anyway when I choose new scope name for new filter there's nothing to sync... Am I missing something, please?
